commands/channel_e2e_test.go

```diff
@@ -206,3 +206,114 @@ func (s *MmctlE2ETestSuite) TestUnarchiveChannelsCmdF() {
+	})
+
+	s.RunForSystemAdminAndLocal("Rename channel", func(c client.Client) {
+		printer.Clean()
+
+		newChannelName := api4.GenerateTestChannelName()
+		newChannelDisplayName := "dn_" + newChannelName
+
+		cmd := &cobra.Command{}
+		cmd.Flags().String("name", newChannelName, "")
+		cmd.Flags().String("display_name", newChannelDisplayName, "")
+
+		err := renameChannelCmdF(c, cmd, []string{s.th.BasicTeam.Id + ":" + channel.Id})
+		s.Require().Nil(err)
+
+		s.Require().Len(printer.GetLines(), 1)
+		printedChannel, ok := printer.GetLines()[0].(*model.Channel)
+		s.Require().True(ok, "unexpected printer output type")
+
+		s.Require().Equal(newChannelName, printedChannel.Name)
+		s.Require().Equal(newChannelDisplayName, printedChannel.DisplayName)
+
+		rchannel, err := s.th.App.GetChannel(channel.Id)
+		s.Require().Nil(err)
+		s.Require().Equal(newChannelName, rchannel.Name)
+		s.Require().Equal(newChannelDisplayName, rchannel.DisplayName)
+	})
+
+	s.Run("Rename channel without permission", func() {
+		printer.Clean()
+
+		channelInit, appErr := s.th.App.GetChannel(channel.Id)
+		s.Require().Nil(appErr)
+
+		newChannelName := api4.GenerateTestChannelName()
+		newChannelDisplayName := "dn_" + newChannelName
+
+		cmd := &cobra.Command{}
+		cmd.Flags().String("name", newChannelName, "")
+		cmd.Flags().String("display_name", newChannelDisplayName, "")
+
+		err := renameChannelCmdF(s.th.Client, cmd, []string{s.th.BasicTeam.Id + ":" + channel.Id})
+		s.Require().NotNil(err)
+		s.Require().Len(printer.GetLines(), 0)
+		s.Require().Len(printer.GetErrorLines(), 0)
+		s.Require().Equal(fmt.Sprintf("cannot rename channel \"%s\", error: : You do not have the appropriate permissions., ", channelInit.Name), err.Error())
+
+		rchannel, err := s.th.App.GetChannel(channel.Id)
+		s.Require().Nil(err)
+		s.Require().Equal(channelInit.Name, rchannel.Name)
+		s.Require().Equal(channelInit.DisplayName, rchannel.DisplayName)
+	})
+
+	s.Run("Rename channel with permission", func() {
+		printer.Clean()
+
+		_, appErr := s.th.App.AddChannelMember(s.th.BasicUser.Id, channel, "", "")
+		s.Require().Nil(appErr)
+
+		newChannelName := api4.GenerateTestChannelName()
+		newChannelDisplayName := "dn_" + newChannelName
+
+		cmd := &cobra.Command{}
+		cmd.Flags().String("name", newChannelName, "")
+		cmd.Flags().String("display_name", newChannelDisplayName, "")
+
+		err := renameChannelCmdF(s.th.Client, cmd, []string{s.th.BasicTeam.Id + ":" + channel.Id})
+		s.Require().Nil(err)
+
+		s.Require().Len(printer.GetLines(), 1)
+		printedChannel, ok := printer.GetLines()[0].(*model.Channel)
+		s.Require().True(ok, "unexpected printer output type")
+
+		s.Require().Equal(newChannelName, printedChannel.Name)
+		s.Require().Equal(newChannelDisplayName, printedChannel.DisplayName)
+
+		rchannel, err := s.th.App.GetChannel(channel.Id)
+		s.Require().Nil(err)
+		s.Require().Equal(newChannelName, rchannel.Name)
+		s.Require().Equal(newChannelDisplayName, rchannel.DisplayName)
+	})
+}
```

Comment:
Please add an empty line here to fix the linter issue

Reply:
done